lint-staged.config.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+const micromatch = require( 'micromatch' );
+
+/**
+ * Internal dependencies
+ */
+const { plugins } = require( './plugins.json' );
+
+/**
+ * Join and escape filenames for shell.
+ *
+ * @param {Array} files Files to join.
+ *
+ * @return {string} Joined files.
+ */
+const joinFiles = ( files ) => {
+	return files.map( ( file ) => `'${ file }'` ).join( ' ' );
+};
+
+module.exports = {
+	'**/*.js': ( files ) => `npm run lint-js ${ joinFiles( files ) }`,
+	'**/*.php': ( files ) => {
+		const commands = [ 'composer phpstan' ];
+
+		plugins.forEach( ( plugin ) => {
+			const pluginFiles = micromatch(
+				files,
+				`**/plugins/${ plugin }/**`
```

Comment:
PhpStorm is complaining here: "Invalid number of arguments, expected 3"

```suggestion
				`**/plugins/${ plugin }/**`,
				{}
```

It seems micromatch isn't properly doing its jsdoc to make the `options` arg optional.

Reply:
Done as a part of https://github.com/WordPress/performance/pull/1090/commits/52d07aef7d9d94a9378d91e10db4368d1e62509f.